src/venue-service/models/terminal/venue-terminal.ts

```diff
@@ -0,0 +1,27 @@
+import { TerminalItem } from './terminal-item';
+import { ApiVenueTerminal } from '../../typings';
+
+export class VenueTerminal {
+  private readonly directions: string;
+  private readonly journeyTime: string;
+  private readonly terminal: TerminalItem;
+
+  constructor (data: ApiVenueTerminal) {
+    this.directions = data.directions;
+    this.journeyTime = data.journeyTime;
+    this.terminal = new TerminalItem(data.terminal);
```

Comment:
Also check that terminal will not be null, because in this case, this line would throw an error

Reply:
I don't see the check for that

Reply:
we have checkRequiredProperty(data, 'VenueTerminal: settings') above. this is how we check if terminal is null or not

Reply:
That's ok for the case when user is trying to create an instance manually, but will break the call if the response will not have terminal details (which could be a real case)

Reply:
done

Reply:
also i removed checkRequiredProperty because in this case we don't need it